### PR TITLE
Fix missing role in helm chart

### PR DESCRIPTION
**Pull request by claudio4j**
Fix https://github.com/apache/camel-k/issues/4629

- Also improve doc about helm and minikube installation

**Release Note**
```release-note
NONE
```

### CODE REVIEW

File: docs/charts/index.yaml [hunk 1]
```diff
@@ -1,9 +1,36 @@
 apiVersion: v1
 entries:
   camel-k:
+  - apiVersion: v2
+    appVersion: 2.1.0-SNAPSHOT
```
Comment (oscerd) [inline docs/charts/index.yaml:5]: I don't think we should publish a snapshot. We should recreate the helm chart for 2.0.0 and push it again with the role fixed.

**Review by oscerd — CHANGES_REQUESTED** [review]
I think we should just recreate the helm chart for 2.0.0 and not upload a snapshot.

File: docs/charts/index.yaml (hunk 1 above)
Comment (claudio4j) [reply docs/charts/index.yaml]: Sure, makes sense, thanks for spotting it.

File: helm/camel-k/templates/operator-role.yaml [hunk 2]
```diff
@@ -111,6 +117,7 @@ rules:
   resources:
   - roles
   - rolebindings
+  - clusterrolebindings
```
Comment (squakez) [inline helm/camel-k/templates/operator-role.yaml:120]: Why is this required?

File: helm/camel-k/templates/operator-role.yaml [hunk 3]
```diff
@@ -167,6 +174,13 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
```
Comment (squakez) [inline helm/camel-k/templates/operator-role.yaml:177]: Why is this required?

File: helm/camel-k/templates/operator-role.yaml [hunk 4]
```diff
@@ -92,6 +97,7 @@ rules:
   - ""
   resources:
   - pods/proxy
+  - pods/log
```
Comment (squakez) [inline helm/camel-k/templates/operator-role.yaml:100]: Why is this required?

**Review by squakez — COMMENTED** [review]
Thanks for taking care. It seems we missed to sync after the changes done in the new naming of `Pipes`

**Comment by claudio4j** [comment]
Pasquale, I forgot to mention the additional apigroup resources, were synced from [operator-role.yaml](https://github.com/apache/camel-k/blob/main/config/rbac/operator-role.yaml)
